tests/Util/PhpServicesCreatorTest.php

```diff
@@ -0,0 +1,175 @@
+<?php
+
+/*
+ * This file is part of the Symfony MakerBundle package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Bundle\MakerBundle\Tests\Util;
+
+use PHPUnit\Framework\TestCase;
+use Symfony\Bundle\MakerBundle\Util\PhpServicesCreator;
+
+// For testing parameters conversion
+\define('GLOBAL_CONSTANT', 'value');
+
+class PhpServicesCreatorTest extends TestCase
+{
+    // For testing parameters conversion
+    const CONSTANT_NAME = 1;
+
+    public function testParametersConversionInEveryCase()
+    {
+        $source = file_get_contents(__DIR__.'/fixtures/YamlConvert/source/parameters.yaml');
+        $expectedSource = file_get_contents(__DIR__.'/fixtures/YamlConvert/parameters.php');
+
+        $creator = new PhpServicesCreator();
+
+        $this->assertSame($expectedSource, $creator->convert($source));
+    }
```

Comment:
You should use a `@dataProvider` for this. Heck, I would probably use the `Finder` object inside a data provider method to look for all `.yaml` files in `YamlConver/source` and convert it into a test case automatically.

Also, to make this test suite even MORE impactful, you could (at the end of the test) process the YAML file through the `YamlFileLoader` and the PHP file through the `PhpFileLoader`. You can pass both an empty `ContainerBuilder` and then compare that the `ContainerBuilders` are equal after the test. This would prove that each creates the same container.

Reply:
yes, good idea for the container. I'm bad for testing, but it's a good opportunity for me to better myself.

Reply:
Whaou! but i need to adapt all the fake configuration! It might take some time to create real use cases with fake Classes, resources etc... Just to warn you.

Reply:
For the dataProvider we code something like this:
`

    public function getConfigurationFilesDetails()
    {
        $finder = Finder::create()->files()->name('*.php')->in(__DIR__.'/fixtures/YamlConvert/')->exclude('FakeClass');
        foreach ($finder as $file) {
            $nameTestCase = str_replace('.php', '', $file->getBasename());

            yield $nameTestCase => [
                $file->getPath(),
                $nameTestCase,
                file_get_contents($file->getPath().'/source/'.$nameTestCase.'.yaml'),
                file_get_contents($file->getPath().'/'.$file->getFilename()),

            ];
        }
    }`

Reply:
And test the convertion like this for example:
`    /**
     * @dataProvider getConfigurationFilesDetails
     */
    public function testConvertionYamlServices(string $folderPath, string $baseFileName, string $yamlFile, string $phpFile)
    {
        $creator = new PhpServicesCreator();
        $this->assertSame($phpFile, $creator->convert($yamlFile));

        $yamlContainerBuilder = new ContainerBuilder();
        $yamlLoader = new YamlFileLoader($yamlContainerBuilder, new FileLocator($folderPath.'/source'));
        $yamlLoader->load($baseFileName.'.yaml');

        $phpContainerBuilder = new ContainerBuilder();
        $phpLoader = new PhpFileLoader($phpContainerBuilder, new FileLocator($folderPath));
        $phpLoader->load($baseFileName.'.php');

        $this->assertSame($yamlContainerBuilder, $phpContainerBuilder);
    }`

Reply:
```
  /**
     * @dataProvider getConfigurationFilesDetails
     */
    public function testConvertionYamlServices(string $folderPath, string $baseFileName, string $yamlFile, string $phpFile)
    {
        $creator = new PhpServicesCreator();
        $this->assertSame($phpFile, $creator->convert($yamlFile));

        $yamlContainerBuilder = new ContainerBuilder();
        $yamlLoader = new YamlFileLoader($yamlContainerBuilder, new FileLocator($folderPath.'/source'));
        $yamlLoader->load($baseFileName.'.yaml');

        $phpContainerBuilder = new ContainerBuilder();
        $phpLoader = new PhpFileLoader($phpContainerBuilder, new FileLocator($folderPath));
        $phpLoader->load($baseFileName.'.php');

        $this->assertSame($yamlContainerBuilder, $phpContainerBuilder);
    }`
```

Reply:
But for the container equality, i think we can't compare them directly because they have specific properties according to the configuration format. But we can compare `->getParameterBag()`, `->getDefinitions()` etc.

Reply:
```
    public function testParameters () {

        $source = file_get_contents(__DIR__.'/fixtures/YamlConvert/source/parameters.yaml');
        $expectedSource = file_get_contents(__DIR__.'/fixtures/YamlConvert/parameters.php');

        $creator = new PhpServicesCreator();

        $this->assertSame($expectedSource, $creator->convert($source));

        $yamlContainerBuilder = new ContainerBuilder();
        $yamlLoader = new YamlFileLoader($yamlContainerBuilder, new FileLocator(__DIR__.'/fixtures/YamlConvert/source'));
        $yamlLoader->load('parameters.yaml');

        $phpContainerBuilder = new ContainerBuilder();
        $phpLoader = new PhpFileLoader($phpContainerBuilder, new FileLocator(__DIR__.'/fixtures/YamlConvert/'));
        $phpLoader->load('parameters.php');

        $this->assertTrue($yamlContainerBuilder->getParameterBag() == $phpContainerBuilder->getParameterBag());
    }

    public function testDefaultConfiguration()
    {
        $yamlContainerBuilder = new ContainerBuilder();
        $yamlLoader = new YamlFileLoader($yamlContainerBuilder, new FileLocator(__DIR__.'/fixtures/YamlConvert/source'));
        $yamlLoader->load('default.yaml');

        $phpContainerBuilder = new ContainerBuilder();
        $phpLoader = new PhpFileLoader($phpContainerBuilder, new FileLocator(__DIR__.'/fixtures/YamlConvert/'));
        $phpLoader->load('default.php');
        // dd($yamlContainerBuilder->getDefinitions(), $phpContainerBuilder->getDefinitions());
        $this->assertTrue($yamlContainerBuilder->getDefinitions() == $phpContainerBuilder->getDefinitions());
    }
```

Reply:
You can check the last commit.